src/amidiplug/amidi-plug.cc

```diff
@@ -150,15 +154,11 @@ bool AMIDIPlug::read_tag (const char * filename, VFSFile & file, Tuple & tuple,
 
     tuple.set_str (Tuple::Codec, "MIDI");
     tuple.set_int (Tuple::Length, mf.length / 1000);
+    tuple.set_int (Tuple::Channels, s_channels);
```

Comment:
read_tag() should not be using a static variable; it needs to be thread-safe.

Reply:
Would explicit call to function **backend_audio_info** be fine?

Reply:
That seems fine.